### PR TITLE
[s3-repository] Don't fail if there no symlink for AWS Web Identity Token

### PR DESCRIPTION
**Pull request by arteam**
Make sure users can use the static credentials even if there is a service account with IAM roles configured on the system.

See https://github.com/elastic/elasticsearch/issues/52625#issuecomment-1060154207

### CODE REVIEW

File: modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java [hunk 1]
```diff
@@ -294,10 +294,12 @@ public void refresh() {
             // Make sure that a readable symlink to the token file exists in the plugin config directory
             Path webIdentityTokenFileSymlink = environment.configFile().resolve("repository-s3/aws-web-identity-token-file");
             if (Files.exists(webIdentityTokenFileSymlink) == false) {
-                throw new IllegalStateException("A Web Identity Token symlink in the config directory doesn't exist");
+                LOGGER.warn("A Web Identity Token symlink in the config directory doesn't exist");
+                return;
             }
             if (Files.isReadable(webIdentityTokenFileSymlink) == false) {
-                throw new IllegalStateException("Unable to read a Web Identity Token symlink in the config directory");
+                LOGGER.warn("Unable to read a Web Identity Token symlink in the config directory");
```
[the remark this answers is not in the document]
Comment (arteam) [reply modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java:301]: There are reports from users who have this environment variable configured, but they do not want to use it (and configure a symlink) and prefer to use static credentials (https://github.com/elastic/elasticsearch/issues/52625#issuecomment-1060154207).

File: modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java [hunk 2]
```diff
@@ -292,9 +292,11 @@ public void refresh() {
                 return;
             }
             // Make sure that a readable symlink to the token file exists in the plugin config directory
+            // If it doesn't, the user doesn't wan to use AWS web identity tokens
```
Comment (tlrx) [inline modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java:295]: Can we change both comment lines to something like:
```suggestion
            // AWS_WEB_IDENTITY_ENV_VAR exists but we only use  Web Identity Tokens if a corresponding symlink exists and is readable
```

File: modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java [hunk 3]
```diff
@@ -292,9 +292,11 @@ public void refresh() {
...
             Path webIdentityTokenFileSymlink = environment.configFile().resolve("repository-s3/aws-web-identity-token-file");
             if (Files.exists(webIdentityTokenFileSymlink) == false) {
-                throw new IllegalStateException("A Web Identity Token symlink in the config directory doesn't exist");
+                LOGGER.warn("A Web Identity Token symlink in the config directory doesn't exist");
```
Comment (tlrx) [inline modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java:298]:
```suggestion
                LOGGER.warn("Cannot use AWS Web Identity Tokens: AWS_WEB_IDENTITY_ENV_VAR is defined but no corresponding symlink exists in the config directory");
```

**Review by tlrx — APPROVED** [review]
LGTM but I left comments to address

**Comment by arteam** [comment]
@elasticmachine upgrade branch

**Comment by arteam** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Comment by arteam** [comment]
Thanks Tanguy!

**Pull request merged**